### PR TITLE
ROX-11192 the collector can be built using podman

### PR DESCRIPTION
**Pull request by ovalenti**
## Description

We want to support podman as the container provider for the build process, and keep compatibility with Docker

## Checklist
- [x] Investigated and inspected CI test results
- [ ] ~~Updated documentation accordingly~~

## Testing Performed
The build image and collector builds fine with this patch

### CODE REVIEW

**Comment by Molter73** [comment]
The change looks good, I'm testing the build right now. Could you add the `build-builder-image` label to the PR and re-trigger CI so we can double check everything works correctly?

File: builder/install/10-protobuf.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 set -e
 
-curl -L "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_VERSION}/protobuf-cpp-${PROTOBUF_VERSION}.tar.gz" | tar xz
+curl -L "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_VERSION}/protobuf-cpp-${PROTOBUF_VERSION}.tar.gz" | tar xz --no-same-owner
```
Comment (erthalion) [inline builder/install/10-protobuf.sh:5]: Could you elaborate more (maybe in the comment, or the commit message) in which way is it related to the podman vs docker compatibility?

Comment (ovalenti) [reply builder/install/10-protobuf.sh:5]: Sure, this is related to running the container as non-root and the corresponding mapping to the actual user id.
If the [commit log](https://github.com/stackrox/collector/commit/f0facf5cc09077ac6efa628e2bea86a1c38e3764) should go further in the details, please tell me and I will elaborate.

**Comment by ovalenti** [comment]
Rebased on master to work on depending PR

**Comment by ovalenti** [comment]
@erthalion I re-triggered the CI pipeline as siggested and it looks like the failing test is related to a failing kernel module build, which is deactivated in _master_
`ERROR: modpost: "cc_mkenc" [/scratch/module-src/1.1.0/collector.ko] undefined!`

Would you like me to rebase to have it all green before approving ?

**Pull request merged**